api/firmware/psbt_test.go

```diff
@@ -0,0 +1,596 @@
+// Copyright 2025 Shift Crypto AG
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package firmware
+
+import (
+	"bytes"
+	"encoding/binary"
+	"encoding/hex"
+	"testing"
+
+	"github.com/BitBoxSwiss/bitbox02-api-go/api/firmware/messages"
+	"github.com/BitBoxSwiss/bitbox02-api-go/util/errp"
+	"github.com/BitBoxSwiss/bitbox02-api-go/util/semver"
+	"github.com/btcsuite/btcd/btcec/v2/schnorr"
+	"github.com/btcsuite/btcd/btcutil"
+	"github.com/btcsuite/btcd/btcutil/psbt"
+	"github.com/btcsuite/btcd/chaincfg"
+	"github.com/btcsuite/btcd/txscript"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/davecgh/go-spew/spew"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/protobuf/proto"
+)
+
+type psbtPrevoutFetcher struct {
+	psbt *psbt.Packet
+}
+
+// FetchPrevOutput implements `txscript.PrevOutputFetcher`.
+func (p psbtPrevoutFetcher) FetchPrevOutput(op wire.OutPoint) *wire.TxOut {
+	for inputIndex := range p.psbt.Inputs {
+		psbtInput := &p.psbt.Inputs[inputIndex]
+		txInput := p.psbt.UnsignedTx.TxIn[inputIndex]
+		if txInput.PreviousOutPoint.String() == op.String() {
+			if psbtInput.WitnessUtxo != nil {
+				return psbtInput.WitnessUtxo
+			}
+			if psbtInput.NonWitnessUtxo != nil {
+				return psbtInput.NonWitnessUtxo.TxOut[txInput.PreviousOutPoint.Index]
+			}
+		}
+	}
+	panic("no prevoutput")
```

Comment:
same as for the other test file, I think having a `t *testing.T` argument and calling `t.Fail` here would be better

Reply:
Can't do that here as this is an interface function. But looking at other impls of the same interface, they just return nil if none found, so I'll do that here too.